src/transformers/visitors/toOrchestrationVisitor.ts

```diff
@@ -189,6 +236,284 @@ const visitor = {
+ stateNode.mappingKey =  stateNode.mappingKey.replace(oldStateName, newStateArray[index])
+}
+}
+}
+})
+newPostStatementList.splice(- 3);
+ }
+})
+  node._newASTPointer.forEach(file => {
+  if(file.fileName === callingFncName[index])
+  {
+  file.nodes.forEach(childNode => {
+  if(childNode.nodeType === 'FunctionDefinition')
+  {
+    childNode.parameters.modifiedStateVariables = [...new Set([...childNode.parameters.modifiedStateVariables, ...newParametersList])];
+    if(childNode.decrementedSecretStates)
+    childNode.decrementedSecretStates = [...new Set([...childNode.decrementedSecretStates, ...newdecrementedSecretStates])];
+    childNode.body.preStatements = [...new Set([...childNode.body.preStatements, ...newPreStatementList])]
+    childNode.body.statements = [...new Set([...childNode.body.statements, ...newStatementList])]
+    const index = childNode.body.postStatements.findIndex((node) => (node.nodeType=== 'CalculateCommitment'));
+
+    childNode.body.postStatements = merge(childNode.body.postStatements, newPostStatementList , index+1);
+    childNode.body.postStatements.forEach(node => {
+    if(node.nodeType === 'GenerateProof'){
+     node.privateStates = Object.assign(node.privateStates,generateProofNode.privateStates)
+     node.parameters = [...new Set([...node.parameters ,...generateProofNode.parameters])];
+    }
+    if(node.nodeType === 'SendTransaction')
+     node.privateStates = Object.assign(node.privateStates,sendTransactionNode.privateStates)
+
+    if(node.nodeType === 'WritePreimage')
+     node.privateStates = Object.assign(node.privateStates,writePreimageNode.privateStates)
+    })
+    }
+  })
+}
+})
+}
+
+
+
+if(circuitImport[index] === 'false')
+{
+  file.nodes.forEach(childNode => {
+    if(childNode.nodeType === 'FunctionDefinition'){
+      newStatementList = cloneDeep(childNode.body.statements);
+      newStatementList.forEach(node => {
+        if(node.nodeType === 'VariableDeclarationStatement')
+        {
+        for(const [index, oldStateName] of  oldStateArray.entries()) {
+        node.initialValue.leftHandSide.name = node.initialValue.leftHandSide.name.replace('_'+oldStateName, '_'+ newStateArray[index]);
+         node.initialValue.rightHandSide.name = node.initialValue.rightHandSide.name.replace(oldStateName,  newStateArray[index]);
+
+          }
+      }
+      })
+      }
+  })
+  node._newASTPointer.forEach(file => {
+  if(file.fileName === callingFncName[index])
+  {
+
+  file.nodes.forEach(childNode => {
+  if(childNode.nodeType === 'FunctionDefinition')
+  {
+  childNode.body.statements.forEach(node => {
+    if(node.nodeType === 'ExpressionStatement' && node.expression.name === internalFncName[index])
+{
+  newStatementList.forEach(list => {
+    if(list.nodeType === 'VariableDeclarationStatement'){
+    node.expression = Object.assign(node.expression,list.initialValue);
+    console.log(node.expression);
+    }
+  })
+}
+
+  });
+    }
+    })
+
+}
+})
+}
+}
+})
+})
```

Comment:
Could we separate this into a function at the top or maybe a new file? The code is great but lengthy!